gcloud/datastore/client.py

```diff
@@ -27,6 +23,89 @@
 from gcloud.datastore._implicit_environ import get_connection
 
 
+_MAX_LOOPS = 128
+"""Maximum number of iterations to wait for deferred keys."""
+
+
+def _extended_lookup(connection, dataset_id, key_pbs,
+                     missing=None, deferred=None,
+                     eventual=False, transaction_id=None):
+    """Repeat lookup until all keys found (unless stop requested).
+
+    Helper function for :method:`Client.get_multi`.
```

Comment:
Double-checked. This is only change from original. (I'm trying to be methodical.)